lib/Doctrine/DBAL/Logging/SQLLogger.php

```diff
@@ -12,9 +14,9 @@ interface SQLLogger
     /**
      * Logs a SQL statement somewhere.
      *
-     * @param string         $sql    The SQL to be executed.
-     * @param mixed[]        $params The SQL parameters.
-     * @param int[]|string[] $types  The SQL parameter types.
+     * @param string                $sql    The SQL to be executed.
+     * @param mixed[]               $params The SQL parameters.
+     * @param int[]|string[]|Type[] $types  The SQL parameter types.
```

Comment:
I believe, here it should be `(int|string|Type)[]`. Unlike the above, the array elements don't have to be of the same type and we used this syntax earlier because it was not supported by PHPStan or PhpStorm. If both are fine which the proposed syntax, let's update the type.

Reply:
PHPStorm doesn't understand it at the moment. What do we do then?

Reply:
Keep it as is.